### PR TITLE
fix: Update sys.exit and SystemExit exception to have the same types

### PR DESCRIPTION
**Pull request by kkirsche**
fix: #8513

This updates the types of `sys.exit` and `SystemExit` with the recommendations of @Akuli. The issue had received 👍 s from three team members, so I wanted to help move this along for the team.

### CODE REVIEW

**Review by AlexWaygood — COMMENTED** [review]
Thanks, this looks good. The mypy_primer hits are a _little_ unfortunate, but they're arguably "true positives", and I can live with them.

Two things that might be good to change, before merging:

1. Presumably `builtins.quit()` and `builtins.exit()` should be changed in the same way as `sys.exit()`?
2. Can we define a type alias in the `sys` module for `str | int | None`, and then use that for all of these functions, and the `code` attribute of `SystemExit`? I think it would help ensure that they don't drift out of sync in the future.

**Comment by kkirsche** [comment]
Added 🙂

File: stdlib/sys.pyi
Comment (AlexWaygood) [inline stdlib/sys.pyi:225]: This comment is now out of date. Can you delete it, and add a comment above the definition of sys._ExitCode, summarising the rationale laid out in https://github.com/python/typeshed/issues/8513#issue-1333671093 for why we're going with `str | int | None` here? (You could maybe just link to that comment in the comment.)

Comment (kkirsche) [reply stdlib/sys.pyi]: Added a link to the comment as I think to do the full rationale could end up being rather verbose

Comment (AlexWaygood) [reply stdlib/sys.pyi]: Looks good :)

**Review by AlexWaygood — APPROVED** [review]
LGTM, but I'll wait a little while before merging to see if any other maintainers have any thoughts :)

**Comment by kkirsche** [comment]
works for me, thanks for the review

**Pull request merged**
